### PR TITLE
11_circular_distribution_csimeone Draft Figures

### PR DESCRIPTION
**Pull request by csimeone-usgs**
@ceenell Here is my current draft of my 2 figures and code for day 11 circular distributions. I plan on attending the review session on Thursday to get some feedback hopefully. Additionally let me know what format is preferred for captions etc....

### CODE REVIEW

**Comment by csimeone-usgs** [comment]
@ceenell, this should be ready to be merged, I'm happy to do so my self tomorrow, but I know some groups have specific rules for merges and who can do them. The figures should be pretty good to go, and I suppose maybe they could have minor changes, but I don't expect them to really change. 
Cheers,

**Comment by cnell-usgs** [comment]
I'll check this out today

File: 11_circular_csimeone/scr/plot_gages.R [hunk 1]
```diff
@@ -0,0 +1,122 @@
+# Plot gage locations
+
+library(tibble)
+library(dplyr)
+library(stringr)
+library("ggplot2")
+library("cowplot")
```
Comment (cnell-usgs) [inline 11_circular_csimeone/scr/plot_gages.R]: I'd suggest using the same syntax here as the other packages to be consistent

File: 11_circular_csimeone/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Circular Distributions of CONUS Wide Streamflow
```
Comment (cnell-usgs) [inline 11_circular_csimeone/README.md:1]: Include a section here describing where the data came from and a short description of the data processing steps that were done

File: 11_circular_csimeone/scr/plot_the_driest_and_wettest_days.R [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+#   #NWIS data pull
+#   station <- site_list[i]
+#   cat(station, i, "of", length(site_list), "\n")
+# 
+#   df <- readNWISdv(station,pCd,start_date,end_date) %>%
+#     renameNWISColumns() %>%
+#     as_tibble()
+# 
+#   write_csv(df, paste0("11_circular_csimeone/data_in/NWIS_data/", station, ".csv"))
+# }
```
Comment (cnell-usgs) [inline 11_circular_csimeone/scr/plot_the_driest_and_wettest_days.R]: same as above, I would clean up the unused pieces that are commented out

File: 11_circular_csimeone/scr/plot_the_driest_and_wettest_days.R [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    
+    df_temp <- read.csv(paste0("11_circular_csimeone/data_in/NWIS_data/", site_list[[i]], ".csv")) %>%
+      as_tibble()
+    # df_temp$site <- site_list[[i]]
+    
+    df_list[[i]] <- df_temp
+  }, error=function(e){cat("Site", i, "ERROR ", conditionMessage(e), "\n")})
+}
+
+df <- bind_rows(df_list) %>%
```
Comment (cnell-usgs) [inline 11_circular_csimeone/scr/plot_the_driest_and_wettest_days.R:52]: Add a comment here describing what this step does

File: 11_circular_csimeone/scr/plot_the_driest_and_wettest_days.R [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+  height = unit(0.6, "in"),
+  hjust = 0, vjust = 1,
+  gp = gpar(fill = "white", alpha = 1, col = 'white')
+)
+
+p_2 <- ggdraw() +
+  draw_plot(p_1 , x=-0.11, y = -0.1, width = 1.2, height = 1.2 ) + 
+  draw_grob(rect) +
+  draw_label("When Are Our Rivers Wet or Dry?", x = 0.5, y = 0.9) +
+  draw_text(fig_text, x = 0.5, y = 0.1, size = 12)
```
Comment (cnell-usgs) [inline 11_circular_csimeone/scr/plot_the_driest_and_wettest_days.R:150]: smart use of `ggdraw` to add annotations

File: 11_circular_csimeone/README.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+This sub directory creates a circular plot with the total amount of water passing all gages 2 gages passing auditing requirements for 1950-2020. 
+
+# Post 1
```
Comment (cnell-usgs) [inline 11_circular_csimeone/README.md]: Here, you could change the header to say "the driest and wettest day of the year" to coincide with the script name and give a description of the chart that it creates. I'd also suggest linking here or specifically naming the script that creates that chart

File: 11_circular_csimeone/README.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+![](viz/Wettest_Day_of_the_Year.png)
+
+The amount of flow in rivers in the continental United States varies throughout the year and across the country, but overall where we have gages we see that late winter/early spring months (March - May) tend to have the highest flows and late summer/early fall months tend to have the lowest flows. Extreme events can occur within these long-term ranges, but can also stray outside of them: e.g. extreme low flows in [November 1952](https://pubs.usgs.gov/wsp/1804/report.pdf#:~:text=The%20rare%20occurrence%20of%20a%20succession%20of%20drought-,much%20of%20the%20southern%20half%20of%20the%20Nation.) or extreme high flows in [June 1972](https://pubs.er.usgs.gov/publication/pp924).
+
+# Follow Up Post 2
```
Comment (cnell-usgs) [inline 11_circular_csimeone/README.md]: same comment for here

File: 11_circular_csimeone/README.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+![](viz/Wettest_Day_of_the_Year_CRB.png)
+
+The Colorado River Basin demonstrates how varied sub-basin flows can be, with wetter conditions driven by snowmelt in the upper part of the basin, and drier conditions influenced more by monsoonal rains in the lower part of the basin. The USGS is currently working to better understand hydrological drought dynamics across the country and recently hosted a stakeholder listening session on [streamflow drought](https://www.drought.gov/webinars/drought-prediction-focus-streamflow-end-user-listening-session). Future listening sessions aimed at engaging a wide range of stakeholders will cover [groundwater drought](https://www.drought.gov/events/drought-prediction-focus-groundwater-end-user-listening-session), water use, ecosystems, and drought prediction.
+
+# Follow Up Post 3
```
Comment (cnell-usgs) [inline 11_circular_csimeone/README.md]: same comment for here

File: 11_circular_csimeone/scr/plot_gages.R [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+# 
+# p <- ggplot() + 
+#   geom_map(data=us, map=us,aes(x=long, y=lat, map_id=region, group=group),
+#            fill="#ffffff", color="#7f7f7f", size=0.5) + 
+#   geom_point(data = loc_df, aes(x=longitude, y=latitude)) + 
+#   coord_map("albers", lat0=39, lat1=45)+ 
+#   # coord_map('conic', lat0=30) +
+#   theme_map() +
+#   theme(legend.position = c(0.75, 0.3), plot.margin=unit(c(0,0,0,00),"mm")) 
+# p
```
Comment (cnell-usgs) [inline 11_circular_csimeone/scr/plot_gages.R]: If none of this is used to make the final charts, delete it from the script

File: 11_circular_csimeone/scr/plot_the_driest_and_wettest_days_CRB.R [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+    
+    df_temp <- read.csv(paste0("11_circular_csimeone/data_in/NWIS_data/CRB/", site_list[[i]], ".csv")) %>%
+      as_tibble()
+    # df_temp$site <- site_list[[i]]
+    
+    df_list[[i]] <- df_temp
+  }, error=function(e){cat("Site", i, "ERROR ", conditionMessage(e), "\n")})
+}
+
+df_ucrb <- bind_rows(df_list) %>%
```
Comment (cnell-usgs) [inline 11_circular_csimeone/scr/plot_the_driest_and_wettest_days_CRB.R]: Include a comment to describe what is happening here

File: 11_circular_csimeone/scr/plot_the_driest_and_wettest_days_CRB.R [hunk 11]
```diff
@@ -0,0 +1,197 @@
...
+                       limits = c(0, 1),
+                       oob = scales::squish) +
+  labs(fill = "mm/d flow") + 
+  geom_segment(data=df_lcrb, aes(x = month_start + 2, y = -.1, xend = month_end - 2, yend = -.1), colour = "black", alpha=0.8, size=1)  +
+  geom_text(data=df_lcrb, aes(x = month_median, y = -.3, label=month_ab)) + 
+  # geom_rect(xmin = 45, ymin = 0, xmax = 47, ymax = 0.45302850, color = "grey", fill = NA, size = 1) +
+  geom_rect(xmin = 7, ymin = 0, xmax = 9, ymax = 0.4696284, color = "grey", fill = NA, size = 1) +
+  geom_rect(xmin = 287, ymin = 0, xmax = 289, ymax = 0.04834458, color = "grey", fill = NA, size = 1) +
+  # geom_segment(aes(x = 10, y = 0.6, xend = 35, yend = 0.18), arrow = arrow(length = unit(0.5, "cm"))) +
+  geom_segment(aes(x = 270, y = .6, xend = 285, yend = .1), arrow = arrow(length = unit(0.5, "cm"))) +
```
Comment (cnell-usgs) [inline 11_circular_csimeone/scr/plot_the_driest_and_wettest_days_CRB.R]: I'm impressed you were able to do those arrows in R!

File: 11_circular_csimeone/scr/plot_gages.R (hunk 1 above)
Comment (csimeone-usgs) [reply 11_circular_csimeone/scr/plot_gages.R]: Done

File: 11_circular_csimeone/scr/plot_the_driest_and_wettest_days.R (hunk 3 above)
Comment (csimeone-usgs) [reply 11_circular_csimeone/scr/plot_the_driest_and_wettest_days.R]: These were commented out so I don't download from NWIS every time. Added if statement to only do so when triggered.

File: 11_circular_csimeone/scr/plot_gages.R (hunk 9 above)
Comment (csimeone-usgs) [reply 11_circular_csimeone/scr/plot_gages.R]: Removed

File: 11_circular_csimeone/scr/plot_the_driest_and_wettest_days_CRB.R (hunk 10 above)
Comment (csimeone-usgs) [reply 11_circular_csimeone/scr/plot_the_driest_and_wettest_days_CRB.R]: Added comment for upper and lower basin

File: 11_circular_csimeone/README.md (hunk 2 above)
Comment (csimeone-usgs) [reply 11_circular_csimeone/README.md:1]: I've added some info here

File: 11_circular_csimeone/README.md (hunk 7 above)
Comment (csimeone-usgs) [reply 11_circular_csimeone/README.md]: Added

File: 11_circular_csimeone/README.md (hunk 6 above)
Comment (csimeone-usgs) [reply 11_circular_csimeone/README.md]: Added

File: 11_circular_csimeone/README.md (hunk 8 above)
Comment (csimeone-usgs) [reply 11_circular_csimeone/README.md]: Added here

**Comment by csimeone-usgs** [comment]
@ceenell I've addressed your comments and added them here. Let me know if you want anything else changed before merging. Cheers,

**Comment by cnell-usgs** [comment]
Looking good!

**Pull request merged**